sycl/test/extensions/usm/usm_alloc_utility.cpp

```diff
@@ -83,7 +83,10 @@ int main() {
     array = (int *)malloc_device(N * sizeof(int), q);
     check_and_free(array, dev, ctxt);
 
-    array = (int *)malloc_device(N * sizeof(int), q, property_list{});
+    array = (int *)malloc_device<int>(
+        N * sizeof(int), q,
```

Comment:
```suggestion
    array = malloc_device<int>(
        N, q,
```

I think you introduced a bug here when you changed the call to use the template form of `malloc_device()`.  In the template form, the first parameter is the count of *elements*, not bytes.  Also, there is no need to cast the return type because it is already a pointer to the element type.